cli/command/node/list.go

```diff
@@ -55,61 +48,27 @@ func runList(dockerCli command.Cli, opts listOptions) error {
 		return err
 	}
 
+	info := types.Info{}
 	if len(nodes) > 0 && !opts.quiet {
 		// only non-empty nodes and not quiet, should we call /info api
-		info, err := client.Info(ctx)
+		info, err = client.Info(ctx)
 		if err != nil {
 			return err
 		}
-		printTable(out, nodes, info)
-	} else if !opts.quiet {
-		// no nodes and not quiet, print only one line with columns ID, HOSTNAME, ...
-		printTable(out, nodes, types.Info{})
-	} else {
-		printQuiet(out, nodes)
 	}
 
-	return nil
-}
-
-func printTable(out io.Writer, nodes []swarm.Node, info types.Info) {
-	writer := tabwriter.NewWriter(out, 0, 4, 2, ' ', 0)
-
-	// Ignore flushing errors
-	defer writer.Flush()
-
-	fmt.Fprintf(writer, listItemFmt, "ID", "HOSTNAME", "STATUS", "AVAILABILITY", "MANAGER STATUS")
-	for _, node := range nodes {
-		name := node.Description.Hostname
-		availability := string(node.Spec.Availability)
-
-		reachability := ""
-		if node.ManagerStatus != nil {
-			if node.ManagerStatus.Leader {
-				reachability = "Leader"
-			} else {
-				reachability = string(node.ManagerStatus.Reachability)
-			}
-		}
-
-		ID := node.ID
-		if node.ID == info.Swarm.NodeID {
-			ID = ID + " *"
+	format := opts.format
+	if len(format) == 0 {
+		if len(dockerCli.ConfigFile().NodesFormat) > 0 && !opts.quiet {
+			format = dockerCli.ConfigFile().NodesFormat
+		} else {
+			format = formatter.TableFormatKey
```

Comment:
Could we just use 
```
format = formatter.TableFormatKey
if len(dockerCli.ConfigFile().NodesFormat) > 0 && !opts.quiet {
    format = dockerCli.ConfigFile().NodesFormat
}
```
Since it seems that most of us do not like `else`? 😄  just a nit.